python/python/res/util/enums/message_level_enum.py

```diff
@@ -9,10 +9,39 @@ class MessageLevelEnum(BaseCEnum):
     LOG_WARNING  = None
     LOG_INFO     = None
     LOG_DEBUG    = None
+    LOG_NOTSET   = None
 
+    @staticmethod
+    def to_enum(val):
+        if val is None:
+            return MessageLevelEnum.LOG_WARNING
```

Comment:
Do we envisage actually using a numerical log level of e.g. 33?

Reply:
Well, the classical loggers like log4j, log4c and Python's logging all accept integer levels for logging.  I don't know if we want that though.